packages/node/src/evm/requests/events.ts

```diff
@@ -15,18 +23,26 @@ export const WITHDRAWAL_REQUEST_TOPICS = [airnodeRrpTopics.WithdrawalRequested];
 
 export const WITHDRAWAL_FULFILLED_TOPICS = [airnodeRrpTopics.WithdrawalFulfilled];
 
-export function isApiCallRequest(log: ethers.utils.LogDescription) {
-  return API_CALL_REQUEST_TOPICS.includes(log.topic);
+export function isApiCallRequest(log: EVMEventLog): log is EVMRequestCreatedLog {
+  return API_CALL_REQUEST_TOPICS.includes(log.parsedLog.topic);
 }
 
-export function isApiCallFulfillment(log: ethers.utils.LogDescription) {
-  return API_CALL_FULFILLED_TOPICS.includes(log.topic);
+export function isTemplateApiRequest(log: EVMEventLog): log is EVMTemplateRequestCreatedLog {
+  return log.parsedLog.topic === airnodeRrpTopics.ClientRequestCreated;
 }
 
-export function isWithdrawalRequest(log: ethers.utils.LogDescription) {
-  return WITHDRAWAL_REQUEST_TOPICS.includes(log.topic);
+export function isFullApiRequest(log: EVMEventLog): log is EVMFullApiRequestCreatedLog {
+  return log.parsedLog.topic === airnodeRrpTopics.ClientFullRequestCreated;
 }
 
-export function isWithdrawalFulfillment(log: ethers.utils.LogDescription) {
-  return WITHDRAWAL_FULFILLED_TOPICS.includes(log.topic);
+export function isApiCallFulfillment(log: EVMEventLog): log is EVMRequestFulfilledLog {
+  return API_CALL_FULFILLED_TOPICS.includes(log.parsedLog.topic);
+}
+
+export function isWithdrawalRequest(log: EVMEventLog): log is EVMWithdrawalRequestLog {
+  return WITHDRAWAL_REQUEST_TOPICS.includes(log.parsedLog.topic);
```

Comment:
@andreogle This is a bit confusing to me. You check for the request type using the event topics and in tests you are checking this e.g. by `log.name === 'ClientRequestCreated'`. Is there a specific reason why not to differentiate between the type of logs using only the name?

We could simplify the code by removing https://github.com/api3dao/airnode/blob/master/packages/node/src/evm/contracts/airnodeRrp.ts#L21

Reply:
Good point. I'm fine with using just the name

Reply:
Actually maybe it's because theoretically you can have log with same name but different params? I tried that it remox in the contract compiled without errors. 

Anyway, that is probably antipattern and we will never use it in airnode?
I definitely do'n't want to fix that in this PR

Reply:
Tracked in https://github.com/api3dao/airnode/issues/331